src/System.Numerics.Vectors/tests/Performance/Vector2/Normalize.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Xunit.Performance;
+
+namespace System.Numerics.Tests
+{
+    public static partial class Perf_Vector2
+    {
+        [Benchmark(InnerIterationCount = VectorTests.DefaultInnerIterationsCount)]
+        public static void NormalizeBenchmark()
+        {
+            var expectedResult = new Vector2(-16777216.0f, 16777216.0f);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                Vector2 result;
+
+                using (iteration.StartMeasurement())
+                {
+                    result = NormalizeTest();
+                }
+
+                VectorTests.AssertEqual(expectedResult, result);
+            }
+        }
+
+        public static Vector2 NormalizeTest()
+        {
+            var result = VectorTests.Vector2Value;
+
+            for (var iteration = 0; iteration < Benchmark.InnerIterationCount; iteration++)
+            {
+                result += Vector2.Normalize(result);
```

Comment:
@DrewScoggins stopped by and we had a quit chat. It was determined a reasonable approach here will be to actually go both routes.

That is, we will have a version of the test that just calls `result = Vector2.Normalize(input)` and a version that calls `result += Vector2.Normalize(result);`.

In this fashion, we will have a test with no overhead except for the actual call and a version that will catch future optimizations (such as completely optimizing away the call). We should be able to catch this by seeing a significant drop in one, but not a corresponding drop in the other.